lib/Search/ElasticSearch/ElasticSearchEngine.php

```diff
@@ -118,15 +119,39 @@ protected function validateQuery(SearchQuery &$query)
      */
     private function createSearchParams($query)
     {
-        $params = [
+        $searchStr = $query->getSearchString();
+
+        // Wildcard character required for Elasticsearch
+        $wildcardBe = "*";
+
+        // Override frontend wildcard character
+        if (isset($GLOBALS['sugar_config']['search_wildcard_char'])) {
+            $wildcardFe = $GLOBALS['sugar_config']['search_wildcard_char'];
+            if ($wildcardFe !== $wildcardBe && strlen($wildcardFe) === 1) {
+                $searchStr = str_replace($wildcardFe, $wildcardBe, $searchStr);
+            }
+        }
+
+        // Add wildcard at the beginning of the search string
+        if (isset($GLOBALS['sugar_config']['search_wildcard_infront']) &&
+            $GLOBALS['sugar_config']['search_wildcard_infront'] === true && $searchStr[0] !== $wildcardBe) {
+            $searchStr = $wildcardBe . $searchStr;
+        }
```

Comment:
@Dillon-Brown @mattlorimer Minor improvement. Maybe we could simplify the above by doing.

```
        $addWildCard = $GLOBALS['sugar_config']['search_wildcard_infront'] ?? false;
        if (addWildCard === true && $searchStr[0] !== $wildcardBe) {
            $searchStr = $wildcardBe . $searchStr;
        }
```

Reply:
Unfortunately, there's no support for null coalescing operator in PHP 5.6 :(.

Reply:
ah.. ok. thanks @Dillon-Brown